### DISCUSSION TITLE
Seleniumbase keeps downloading chromedriver while it exists

### DISCUSSION THREAD

**Discussion by Kamran-ov** [seleniumbase/SeleniumBase#2541] [Q&A] [answered]
Hi,  

Since last week, we have been facing weird behavior of the seleniumbase Driver

Although it has the driver already downloaded for the Chrome version,
It keeps on downloading it for each time I create a driver object. 
Not only that but also downloads 2 times the driver for each object creation.

I have just upgraded the seleniumbase to the latest version but it keeps on doing the same.
Could you please suggest what we can do for it not to keep on downloading the new version?

![image](https://github.com/seleniumbase/SeleniumBase/assets/50695352/1f29d09a-1974-40a5-a370-15ccab501ae2)

**Comment by mdmintz** [comment, marked as answer]
You have to upgrade to a newer version of `seleniumbase`. The Chromium Team changed the download origin (https://github.com/seleniumbase/SeleniumBase/issues/2497).

Here's how the full thing looks (on my Mac):

```python
*** chromedriver to download = 122.0.6261.69 (Latest Stable) 

Downloading chromedriver-mac-x64.zip from:
https://storage.googleapis.com/chrome-for-testing-public/122.0.6261.69/mac-x64/chromedriver-mac-x64.zip ...
Download Complete!

Extracting ['chromedriver'] from chromedriver-mac-x64.zip ...
Unzip Complete!

The file [uc_driver] was saved to:
/Users/michael/github/SeleniumBase/seleniumbase/drivers/uc_driver

Making [uc_driver 122.0.6261.69] executable ...
[uc_driver 122.0.6261.69] is now ready for use!
```

To find out what version you have, run this script inside your code to see your actual `seleniumbase` version:

```python
from seleniumbase import __version__
print(__version__)
```

**Comment by slvamegestic** [comment]
Hi @mdmintz, I was also facing the same issue and it seems chromedriver / uc_driver was saved in seleniumbase base  drivers directory

<img width="921" alt="Screenshot 2024-02-29 at 12 44 16 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/35224544/cd43a037-119d-4d99-b4c7-f2be51fcbe15">

but still it was downloading drivers for each run

**Reply by mdmintz** [reply]
Upgrade to the latest `seleniumbase`.
